src/averageAges.js

```diff
@@ -1,61 +1,79 @@
 'use strict';
 
 /**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
  *
  * @param {object[]} people
  * @param {number} century - optional
  *
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const findMen = people.filter(item => item.sex === 'm');
+
+  const filteredAges = findMen.filter((person) => {
+    return century ? Math.ceil(person.died / 100) === century : true;
+  });
+
+  const result1 = filteredAges.reduce((prev, { born, died }) => {
```

Comment:
```suggestion
  const totalAge = filteredAges.reduce((prev, { born, died }) => {
```

use more descriptive names